tokio-threadpool/src/sender.rs

```diff
@@ -84,6 +84,30 @@ impl Sender {
         tokio_executor::Executor::spawn(&mut s, Box::pin(future))
     }
 
+    /// Like `spawn`, but in case of error the future is returned instead of
+    /// dropped
+    pub fn spawn_with_future_back_on_error(
+        &self,
+        future: Pin<Box<dyn Future<Output = ()> + Send>>,
```

Comment:
I would probably take a `T: Future` here and box it after the `prepare_for_spawn` check.

Reply:
You're totally right. I guess I suggested this because for my use case I already had a `Box`ed closure and did not want it to be `Box`ed again.